### PR TITLE
Configure Environments in Application Pipeline View, UI Fixes & Enhancements.

### PR DESCRIPTION
**Pull request by divakarkonakalla**
1.Integrated API to get available environments to show in Left panel in Configure Pipeline view page.
2.Added Docker Icon in instance table view along with logo to indicate the instances which are associated with docker.
3.Added new images & block in blueprint card to show the provider type in blueprint card.
4.Review comments for #129 has taken care.
5.UI issue with pagination bar coming on top of tree is resolved.
6.Added new icon for environment & included it in chef component.
7.resolved icon color related issues.

### CODE REVIEW

File: client/cat3/src/partials/sections/dashboard/workzone/application/popups/configurePipelineViewCtrl.js [hunk 1]
```diff
@@ -8,38 +8,79 @@
...
-				} else {
-					data = response;
-					console.log(data);
-				}
-				optionList = data;
-				for(var i=0;i<data.length;i++) {
-					var envIds = data[i].envId;
-					console.log(envIds);
+			$scope.chefServerID = '';
+			var list, selectedElements, factory, compositeSelector;
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/popups/configurePipelineViewCtrl.js]: This scope variable is not required

File: client/cat3/src/partials/sections/dashboard/workzone/application/popups/configurePipelineViewCtrl.js [hunk 2]
```diff
@@ -8,40 +8,86 @@
...
-					data = response.data;
-				} else {
-					data = response;
-					console.log(data);
+		.controller('configurePipelineViewCtrl', ['$scope', '$modalInstance', 'workzoneServices', 'chefSelectorComponent', 'responseFormatter', 'items', '$q', '$timeout', 
+		function($scope, $modalInstance, workzoneServices, chefSelectorComponent, responseFormatter, items, $q, $timeout) {
+			var selectedElements = [];
+			var newEnvList = [];
+			workzoneServices.getUpdatedEnvConfig(items).then(function (response){
+				newEnvList = response.data[0].envSequence;
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/popups/configurePipelineViewCtrl.js]: The two api calls should be both promises. Loader should be shown till both are resolved.

Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/popups/configurePipelineViewCtrl.js]: This can be further improved to get environments from tree instead of api

File: client/cat3/src/partials/sections/dashboard/workzone/application/popups/configurePipelineViewCtrl.js [hunk 3]
```diff
@@ -8,40 +8,86 @@
...
+			    	var newEnv=[];
+			    	angular.forEach(compositeSelector.getSelectorList(),function(val){
+			    	newEnv.push(val.value);
+			    });
+			    console.log(newEnv);
+			    var envList = {
+					"appDeployPipelineData": {
+						"loggedInUser": "",
+						"projectId": "b38ccedc-da2c-4e2c-a278-c66333564719",
+						"envId": newEnv,
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/popups/configurePipelineViewCtrl.js]: projectId sample value needs to be removed and made dynamic

File: client/cat3/src/partials/sections/dashboard/workzone/application/popups/configurePipelineViewCtrl.js [hunk 4]
```diff
@@ -8,40 +8,86 @@
...
+					if(error.responseText){
+						$scope.errorMessage = error.responseText;
+					}
+				});
+				$modalInstance.close({envList: envList.appDeployPipelineData.envSequence});
+			    },
 				cancel: function() {
-					$modalInstance.dismiss('cancel');
+					$modalInstance.close('cancel');
 				}
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/application/popups/configurePipelineViewCtrl.js]: Revert this change, use dismiss to go to reject handler which expects no configuration changed.

File: client/cat3/src/partials/sections/dashboard/workzone/instance/instanceCtrl.js [hunk 5]
```diff
@@ -72,7 +73,8 @@
 			data:'tabData',
 				columnDefs : [
 					{ name:'Logo', enableSorting: false ,  cellTemplate:'<img src="/cat3/images/global/chef-import.png" ng-show="row.entity.chef"/>'+
-					'<img src="/cat3/images/global/chef-import.png" ng-show="row.entity.puppet"/>', cellTooltip: true},
+					'<img src="/cat3/images/global/chef-import.png" ng-show="row.entity.puppet"/>'+
+					'<img class="docker-image dockerenabledinstacne" whatever="2" alt="Docker" src="images/global/docker.png" ng-show="row.entity.docker"/>', cellTooltip: true},
 					{ name:'Name', field: 'name', cellTemplate:'<span>{{row.entity.name}}</span>'+
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/instance/instanceCtrl.js]: What is whatever="2"? It is there in the older jQuery markup also

File: client/cat3/src/partials/sections/dashboard/workzone/instance/instanceCtrl.js [hunk 6]
```diff
@@ -129,6 +131,10 @@
 					$timeout(function() {
 						$scope.instancesGridOptions.totalItems = $scope.totalCards = result.data.metaData.totalRecords;
 						$scope.tabData = $scope.instanceList = result.data.instances;
+						console.log($scope.instanceList.length);
+					   	if($scope.instanceList.length >= 10) {
+					   		$scope.cardsAvailable = true;
```
Comment (revinjacob) [inline client/cat3/src/partials/sections/dashboard/workzone/instance/instanceCtrl.js]: Check against pagination default pageSize in scope rather than hardcoding to 10.

**Comment by divakarkonakalla** [comment]
@revinjacob , All the review comments has taken care. Please merge this pull request.
